src/coverage-system/renderer.ts

```diff
@@ -116,22 +116,19 @@ export class Renderer {
         if (!section || !section.lines) {
             return;
         }
-        // TODO cleanup this area by using maps, filters, etc
-        section.lines.details.forEach((detail) => {
-            if (detail.line < 0) { return; }
-            const lineRange = new Range(detail.line - 1, 0, detail.line - 1, 0);
-            if (detail.hit > 0) {
-                if (coverageLines.none.find((range) => range.isEqual(lineRange))) {
-                    // remove all none coverage, for this line, if one full exists
-                    coverageLines.none = coverageLines.none.filter((range) => !range.isEqual(lineRange));
-                }
-                coverageLines.full.push(lineRange);
-            } else {
-                const fullExists = coverageLines.full.find((range) => range.isEqual(lineRange));
-                if (!fullExists) {
-                    // only add a none coverage if no full ones exist
-                    coverageLines.none.push(lineRange);
-                }
+        section.lines.details.filter((detail) => detail.line >= 0).forEach((detail) => {
```

Comment:
Similar here as below (around re-ability and a small tweak to the indent)
```ts
  section.lines.details
  .filter((detail) => detail.line >= 0)
  .forEach((detail) => {
      const lineRange = new Range(detail.line - 1, 0, detail.line - 1, 0);
      if (detail.hit > 0) {
          // Evaluates to true if at least one element in range is equal to LineRange
          if (coverageLines.none.some((range) => range.isEqual(lineRange))) {
              coverageLines.none = coverageLines.none.filter((range) => !range.isEqual(lineRange))
          }
          coverageLines.full.push(lineRange);
      } else {
          if (!coverageLines.full.some((range) => range.isEqual(lineRange))) {
              // only add a none coverage if no full ones exist
              coverageLines.none.push(lineRange);
          }
      }
  });
```

Reply:
Makes sense, thanks for the feedback Ryan. I'll make those changes right away.

Reply:
Sweet, thanks! I will try to take a look within the week.